Makefile

```diff
@@ -167,21 +172,43 @@ run-tmpnb-pool: token-check
 	fi;
 
 kill-tmpnb:
-	@-docker rm -f $(TMPNB_PROXY_CONTAINER_NAME) $(TMPNB_POOL_CONTAINER_NAME)
-	@-docker rm -f $$(docker ps -a | grep 'tmp.' | awk '{print $$1}') 2> /dev/null
+	@-docker rm -f $(TMPNB_PROXY_CONTAINER_NAME) $(TMPNB_POOL_CONTAINER_NAME) 2> /dev/null || true
+	@-docker rm -f $$(docker ps -a | grep 'tmp.' | awk '{print $$1}') 2> /dev/null || true
 
 ###### end tmpnb
 
 kill: kill-tmpnb
-	-@docker rm -f $(DASHBOARD_CONTAINER_NAME) $(KG_CONTAINER_NAME)
+	@-docker rm -f $(DASHBOARD_CONTAINER_NAME) $(KG_CONTAINER_NAME) 2> /dev/null || true
 
+test: CMD?=test
+test: SERVER_NAME?=$(DASHBOARD_CONTAINER_NAME)
+test: DOCKER_OPTIONS?=
 test: build
-		@docker run -it --rm \
-			--name $(DASHBOARD_CONTAINER_NAME) \
-			$(DASHBOARD_IMAGE_NAME) test
+	@docker run -it --rm \
+		--name $(SERVER_NAME) \
+		$(DOCKER_OPTIONS) \
+		$(DASHBOARD_IMAGE_NAME) $(CMD)
+
+integration-test: SERVER_NAME?=integration-test-server
+integration-test: IP?=$$(docker-machine ip $$(docker-machine active))
+integration-test: KG_PORT?=8888
+integration-test: KG_AUTH_TOKEN?=1a2b3c4d5e6f
+integration-test: | kill build run-kernel-gateway
+	$(DOCKER_APP) -d \
+		-e KERNEL_GATEWAY_URL=http://$(KG_CONTAINER_NAME):8888 \
+		-e KG_AUTH_TOKEN=$(KG_AUTH_TOKEN) \
+		--link $(KG_CONTAINER_NAME):$(KG_CONTAINER_NAME) \
+		$(DASHBOARD_IMAGE_NAME)
+	@echo 'Waiting 30 seconds for server to start...'
```

Comment:
Can probably shorten this quite a bit because there's another docker build that happens right after it.